thunder/core/jit_ext.py

```diff
@@ -595,6 +595,9 @@ def proxify(self, value: WrappedValue) -> Any:
             co: CACHE_OPTIONS = get_cache_option()
             if co is CACHE_OPTIONS.CONSTANT_VALUES:
                 self.add_constraint((clang.check_tensor_shape_and_metadata, p_orig))
+            elif co is CACHE_OPTIONS.SYMBOLIC_VALUES:
```

Comment:
What is our medium term plan w.r.t. defaul caching? If we need this for correctly handling #231 , it would seem that symbolic values should be the default but that in turn would mean that we want to have it work for our supported use cases.

Reply:
I'm hoping that #231 wouldn't need to the whole `enable CACHE_OPTIONS.SYMBOLIC_VALUES` thing. Looks like #231 has passed all CI, which feels promising.

The first step of this is to get number proxies to be plumbed through, I think that might be helpful for #231.
In terms of dynamic shape, we can slowly expanding its support.

Start off with allowing scalar input as number proxies, while still requiring tensor to be constant shape. I'm trying to figure out how/where to properly insert prologue_trace guard. Right now considering doing that from executor. i.e. nvfuser would require reduction dim(s) to be baked in as constant, while torchex doesn't care.

I'll try to come up with a design doc for review.